kubevirt/node-maintenance-operator
# Operator SDK update

Short title, long story: in order to be future proof and getting rid of old technical debt the project is re-created using the latest operator-sdk (versiom 1.12.0).

The process in simple steps was:
- delete all but e2e tests
- create project
- merge code and config back into new project

For the detailed steps see commit history.
Last commit adds vendor dir, reviewers might want to checkout the code and skip that commit when reviewing... :) 

Things being left out while merging back:
- must gather image: re-add this only when needed
- e2e test on KubeVirtCI: this was using a very old version, and adds much complexity. Not sure anymore if it's worth it, or if e2e on OpenShiftCI is enough?
- old manifests: they were needed in an older OLM package format, today not anymore. For getting older versions just check git history
- deployment manifests: use `operator-sdk run bundle ...` instead

TODOs in a follow up:
- remove e2e job config from KubeVirtCI and then the `automation` dir (it's a no-op now)
- update Dockerfile locations in OpenshiftCI and then remove `build` dir
- test git tag based upstream release
- make sure HCO still works with this (it should, see relevant commit)
- update downstream build config

```release-note
Re-created project with latest operator-sdk 1.12.0 and get rid of technical debt
```
